apiserver/common/networkingcommon/linklayer.go

```diff
@@ -0,0 +1,157 @@
+package networkingcommon
+
+import (
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+	"gopkg.in/mgo.v2/txn"
+
+	"github.com/juju/juju/core/network"
+)
+
+// LinkLayerDevice describes a single layer-2 network device.
+type LinkLayerDevice interface {
+	// MACAddress is the hardware address of the device.
+	MACAddress() string
+
+	// Name is the name of the device.
+	Name() string
+
+	// ProviderID returns the provider-specific identifier for this device.
+	ProviderID() network.Id
+
+	// SetProviderIDOps returns the operations required to set the input
+	// provider ID for the link-layer device.
+	SetProviderIDOps(id network.Id) ([]txn.Op, error)
+}
+
+// LinkLayerAddress describes a single layer-3 network address
+// assigned to a layer-2 device.
+type LinkLayerAddress interface {
+	// DeviceName is the name of the device to which this address is assigned.
+	DeviceName() string
+
+	// Value returns the actual IP address.
+	Value() string
+
+	// Origin indicates the authority that is maintaining this address.
+	Origin() network.Origin
+
+	// SetProviderIDOps returns the operations required to set the input
+	// provider ID for the address.
+	SetProviderIDOps(id network.Id) ([]txn.Op, error)
+
+	// SetOriginOps returns the transaction operations required to change
+	// the origin for this address.
+	SetOriginOps(origin network.Origin) []txn.Op
+
+	// SetProviderNetIDsOps returns the transaction operations required to ensure
+	// that the input provider IDs are set against the address.
+	SetProviderNetIDsOps(networkID, subnetID network.Id) []txn.Op
+}
+
+// LinkLayerMachine describes a machine that can query
+// and update link-layer device data.
+type LinkLayerMachine interface {
+	// AllLinkLayerDevices returns all currently known
+	// layer-2 devices for the machine.
+	AllLinkLayerDevices() ([]LinkLayerDevice, error)
+
+	// AllAddresses returns all IP addresses assigned to the machine's
+	// link-layer devices
+	AllAddresses() ([]LinkLayerAddress, error)
+
+	// AssertAliveOp returns a transaction operation for asserting
+	// that the machine is currently alive.
+	AssertAliveOp() txn.Op
```

Comment:
This method seems to violate SRP as its purpose is different than the other methods in this interface. Any particular reason why this is not defined in a separate interface and then composed with this one in a super interface?